lib/server.js

```diff
@@ -771,12 +771,36 @@ class WebThingServer {
 
   /**
    * Stop listening.
+   *
+   * @param {boolean?} force - Whether or not to force shutdown immediately.
+   * @returns {Promise} Promise which resolves once the server is stopped.
    */
-  stop() {
+  stop(force = false) {
+    const promises = [];
+
     if (this.mdns) {
-      this.mdns.stop();
+      promises.push(new Promise((resolve, reject) => {
+        this.mdns.stop(force, (error) => {
+          if (error) {
+            reject(error);
+          } else {
+            resolve();
+          }
+        });
+      }));
     }
-    this.server.close();
+
+    promises.push(new Promise((resolve, reject) => {
+      this.server.close((error) => {
```

Comment:
This is awesome. 👍
Right, `http` from NodeJS has a callback as well.
https://nodejs.org/api/net.html#net_server_close_callback